src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl

```diff
@@ -103,42 +99,46 @@ function Models.make_ode_function(model::Nonhydrostatic2DModel)
         ρuh = Ym.ρuh
         ρw = Ym.ρw
         ρθ = Ym.ρθ
-
+    
+        # calculate relevant thermodynamic quantities
+        ts = @. TD.PhaseDry_ρθ(model.parameters, ρ, ρθ / ρ)
```

Comment:
I think all these function in Thermodynamics.jl is dispatched upon an abstract type defined in CLIMAParameters. Do we want to do something similar in ClimaAtmos (e.g., specify the type in the Model struct )?

Reply:
I think so